docs/source/guides/schema-design.md

```diff
@@ -0,0 +1,290 @@
+
+But what if we wanted to update more than just a single or couple attributes on a user? Passing each thing we need as a single argument would get tedious. Especially if multiple mutations used similar fields. For this, we can use input types, which are explained in the next section.
+
+<h3 id="mutation-input-types">Input types</h3>
+
+Input types are a special type in GraphQL which are defined as arguments to queries and, more commonly, mutations.  They can be thought of as object types for arguments, in addition to the other scalar types.  Input types are especially useful when multiple mutations require similar information; for example, when creating a user and updating a user require the same fields, like `age` and `name`.
+
+Input types are used like any other type and defining them is similar to a typical object type definitions, but with the `input` keyword rather than `type`.
+
+Here is an example of two mutations that operate on a `User`, _without_ using input types:
+
+```
+type Mutation {
+  createUser(name: String, age: Int, address: String, phone: String): User
+  updateUser(id: ID!, name: String, age: Int, address: String, phone: String): User
+}
+```
+
+To avoid the repetition of argument fields, this can be refactored to use an input type, as follows:
+
+```
+type Mutation {
+  createUser(user: UserInput): User
+  updateUser(id: ID!, user: UserInput): User
+}
+
+input UserInput {
+  name: String
+  age: Int
+  address: String
+  phone: String
+}
+```
+
+<h3 id="mutation-responses">Responses</h3>
+
+Mutations have a higher chance of causing errors than queries since they are modifying data.  A common way to handle errors during a mutation is to simply `throw` an error.  While that's fine, throwing an error in the resolver will return an error to the caller and prevent a partial response, which could be useful in the event of a partial update.  Consider the following mutation example, which tries to update a user's `name` and `age`:
+
+```graphql
+mutation updateUser {
+  updateUser(id: 1, user: { age: -1, name: "Foo Bar" }){
+    name
+    age
+  }
+}
+```
+
+With validation in place, this mutation might cause an error since the `age` is a negative value.  While it’s possible that the entire operation should be stopped, there’s an opportunity to partially update the user’s record with the new `name` and return the updated record with the `age` left untouched.
```

Comment:
My feeling is that it might be really confusing if we recommend people have mutations that can be partially successful. Is there any other reason to have mutations return this kind of status information?